.github/workflows/main.yml

```
@@ -82,3 +111,68 @@
+        uses: appleboy/ssh-action@v0.1.8
+        with:
+          host: ${{ secrets.SERVER_HOST }}
+          username: ${{ secrets.SERVER_USER }}
+          password: ${{ secrets.SERVER_PASSWORD }}
+          script_timeout: 1800  # ⏱️ 30 minute timeout for deployment
+          script: |
+            cd /root/Learnathon/earendel
+            echo "Switching to main branch and pulling latest changes..."
+            timeout 5m git checkout main || exit 1
+            timeout 5m git pull origin main || exit 1
+            
```

YAMLlint (1.35.1)

[error] trailing spaces

(trailing-spaces)